test/s3select_test.cpp

```diff
@@ -238,3 +238,32 @@ TEST(TestS3SElect, utcnow_function)
     ASSERT_EQ(s3select_res, boost::posix_time::to_simple_string(now));
 }
 
+void generate_csv(std::string& out, size_t size) {
+  // schema is: int, float, string, string
+  std::stringstream ss;
+  for (auto i = 0U; i < size; ++i) {
+    ss << i << "," << i/10.0 << "," << "foo"+std::to_string(i) << "," << std::to_string(i)+"bar" << std::endl;
+  }
+  out = ss.str();
+}
+
+TEST(TestS3selectFunctions, avg)
```

Comment:
that's ok
its positive test, on constant values. (will stay with this)

generally, it need also for negative test (for testing the error flow), 
and also on random input.
future task.